src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -101,6 +103,17 @@ public override void Initialize(AnalysisContext analysisContext)
                     {
                         instantiatedTypes.TryAdd(type.BaseType, null);
                     }
+
+                    // Consider class type declared in the CoClass attribute as instantiated
+                    if (type.TypeKind == TypeKind.Interface &&
+                        type.GetAttributes().FirstOrDefault(x => x.AttributeClass.Equals(coClassAttributeSymbol)) is AttributeData coClassAttribute &&
```

Comment:
I am a bit bothered by the perf of getting the attributes for all interfaces but I don't know if there is a better way.

Reply:
I think it is fine - quite a few analyzers would request attributes, and your search is limited to only interfaces.